static/js/stream_popover.js

```diff
@@ -500,6 +515,41 @@ exports.register_topic_handlers = function () {
 
         e.stopPropagation();
     });
+
+    $('body').on('click', '.sidebar-popover-move-topic-messages', function (e) {
+        const topic_row = $(e.currentTarget);
+        const stream_id = topic_row.attr('data-stream-id');
+        const topic_name = topic_row.attr('data-topic-name');
+        const message = home_msg_list.all_messages()
+            .find(m => m.topic === topic_name && m.stream_id === +stream_id);
```

Comment:
I think https://github.com/zulip/zulip/pull/14382 adds the API we need here.

Reply:
... But that API fundamentally can't be completely accurate.  OK, I think we should just do a `GET /messages` to fetch a single message from the topic, and use that; that'll be 100% reliable, with minimal performance impact.

We can pass `anchor=latest`, `num_before=0`, `num_after=0`, and make it really simple.

Reply:
Actually, never mind, that #14382 is irrelevant, as it has per-stream values.  The existing `message_id` elements in `stream_topic_history` have the data that's available in the frontend, but it's not fully accurate.

Reply:
OK, I merged 735785f9852cd44c3715d3fffc17ab0c3525a16b; the comment there is helpful.  I think the approach I'm most excited about here is:

(1) Moving the message_id construction logic to happen when the user hits confirm; there's no reason to do it to build the popover.
(2) Either creating a frontend function to fetch the latest message ID in a given topic the user has access to, and chaining that to have its success handler finish the request; or perhaps better, just adding a small backend API endpoint to rename the topic that basically just calls the `update_message_backend` view function after looking up the right message ID to use.  The latter might be cleaner.

Reply:
For (2), I went with doing thing in the frontend, since you said, next step would be to implement this for a message.